http/sys_rekey.go

```diff
@@ -231,23 +235,161 @@ func handleSysRekeyUpdate(core *vault.Core, recovery bool) http.Handler {
 	})
 }
 
+func handleSysRekeyVerify(core *vault.Core, recovery bool) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		standby, _ := core.Standby()
+		if standby {
+			respondStandby(core, w, r.URL)
+			return
+		}
+
+		repState := core.ReplicationState()
+		if repState.HasState(consts.ReplicationPerformanceSecondary) {
+			respondError(w, http.StatusBadRequest,
+				fmt.Errorf("rekeying can only be performed on the primary cluster when replication is activated"))
+			return
+		}
+
+		ctx, cancel := core.GetContext()
+		defer cancel()
+
+		switch {
+		case recovery && !core.SealAccess().RecoveryKeySupported():
+			respondError(w, http.StatusBadRequest, fmt.Errorf("recovery rekeying not supported"))
+		case r.Method == "GET":
+			handleSysRekeyVerifyGet(ctx, core, recovery, w, r)
+		case r.Method == "POST" || r.Method == "PUT":
+			handleSysRekeyVerifyPut(ctx, core, recovery, w, r)
+		case r.Method == "DELETE":
+			handleSysRekeyVerifyDelete(core, recovery, w, r)
+		default:
+			respondError(w, http.StatusMethodNotAllowed, nil)
+		}
+	})
+}
+
+func handleSysRekeyVerifyGet(ctx context.Context, core *vault.Core, recovery bool, w http.ResponseWriter, r *http.Request) {
+	barrierConfig, barrierConfErr := core.SealAccess().BarrierConfig(ctx)
+	if barrierConfErr != nil {
+		respondError(w, http.StatusInternalServerError, barrierConfErr)
+		return
+	}
+	if barrierConfig == nil {
+		respondError(w, http.StatusBadRequest, fmt.Errorf("server is not yet initialized"))
+		return
+	}
+
+	// Get the rekey configuration
+	rekeyConf, err := core.RekeyConfig(recovery)
+	if err != nil {
+		respondError(w, err.Code(), err)
+		return
+	}
+	if rekeyConf == nil {
+		respondError(w, http.StatusBadRequest, errors.New("no rekey configuration found"))
+		return
+	}
+
+	// Get the progress
+	progress, err := core.RekeyVerifyProgress(recovery)
+	if err != nil {
+		respondError(w, err.Code(), err)
+		return
+	}
+
+	// Format the status
+	status := &RekeyVerificationStatusResponse{
+		Nonce:    rekeyConf.VerificationNonce,
+		T:        rekeyConf.SecretThreshold,
+		N:        rekeyConf.SecretShares,
+		Progress: progress,
+	}
+	respondOk(w, status)
+}
+
+func handleSysRekeyVerifyDelete(core *vault.Core, recovery bool, w http.ResponseWriter, r *http.Request) {
+	err := core.RekeyVerifyRestart(recovery)
+	if err != nil {
+		respondError(w, err.Code(), err)
+		return
+	}
+
+	ctx, cancel := core.GetContext()
```

Comment:
Why do we grab `ctx` here as opposed to having it passed in like the others?

Reply:
Dunno. I can change it here and in the normal delete method. This is mostly c&p, so C&P weird stuff too.